metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_rest_api.py

```diff
@@ -57,9 +61,75 @@ def _create_session(self) -> requests.Session:
     def get_connection_details_by_id(
         self, connection_id: str
     ) -> FivetranConnectionDetails:
-        """Get details for a specific connection."""
-        connection_details = self._session.get(
-            f"{self.config.base_url}/v1/connections/{connection_id}",
-            timeout=self.config.request_timeout_sec,
-        )
-        return FivetranConnectionDetails(**connection_details.json().get("data", {}))
+        """
+        Get details for a specific connection from the Fivetran API.
+
+        Args:
+            connection_id: The Fivetran connection ID to fetch details for.
+
+        Returns:
+            FivetranConnectionDetails: The parsed connection details.
+
+        """
+        try:
+            connection_details = self._session.get(
+                f"{self.config.base_url}/v1/connections/{connection_id}",
+                timeout=self.config.request_timeout_sec,
+            )
+
+            # Check for HTTP errors and raise HTTPError if needed
+            connection_details.raise_for_status()
+
+            response_json = connection_details.json()
+
+            # Check response code at top level (e.g., "code": "Success")
+            response_code = response_json.get("code")
+            if response_code and response_code.lower() != "success":
+                raise ValueError(
+                    f"Response code is not 'success' for connection_id {connection_id}. "
+                    f"Code: {response_code}, Response: {response_json}"
+                )
+
+            data = response_json.get("data", {})
+            if not data:
+                raise ValueError(
+                    f"Response missing 'data' field for connection_id {connection_id}"
+                )
+
+            # Manually extract and construct nested models with only required fields
+            try:
+                config_data = data.get("config", {})
+                config = FivetranConnectionConfig(
+                    auth_type=config_data.get("auth_type"),
+                    sheet_id=config_data.get("sheet_id"),
+                    named_range=config_data.get("named_range"),
+                )
+
+                # Parse into FivetranConnectionDetails with manually extracted fields
+                return FivetranConnectionDetails(
+                    id=data.get("id"),
+                    group_id=data.get("group_id"),
+                    service=data.get("service"),
+                    created_at=data.get("created_at"),
+                    succeeded_at=data.get("succeeded_at"),
+                    paused=data.get("paused"),
+                    sync_frequency=data.get("sync_frequency"),
+                    config=config,
+                )
+            except Exception as e:
```

Comment:
If we drop the manual extraction of fields and solve this via pydantic model validation, this should be a `ValidationError`.


By catching it and converting to ValueError, you lose all that structured info and just get a string message. While from e preserves the exception chain, callers who want to handle Pydantic validation errors specifically can't easily do that anymore.

When Pydantic raises a ValidationError, it includes structured details like:
- Which specific fields failed validation
- What the validation rule was (e.g., "field required", "invalid type")
- Nested field paths for complex objects